backend/src/apiserver/server/util.go

```diff
@@ -46,6 +46,25 @@ func GetPipelineName(queryString string, fileName string) (string, error) {
 	return pipelineName, nil
 }
 
+// This method extract the common logic of the behaviour for updating the default pipeline version.
+// API caller can either explicitly decide through query string ?updatedefaultversion=false
+// or API server will use true by default.
+func GetDefaultVersionUpdate(queryString string) (bool, error) {
```

Comment:
It seems you can replace this function with https://github.com/kubeflow/pipelines/blob/fffe830ba9b1ce0b1bb1ae910fcbb9b572e11baa/backend/src/apiserver/common/config.go#L87

because you won't need decoding to know whether a string is true or not, all related characters do not change when urlencoded

The error handling part is tricky, I agree explicitly report errors are better. You can decide on this.

Reply:
Sound great, will remove it!

Reply:
I realised this will not work if we should allow for users to also set default using an env parameter. Since we need to check if the env parameter is set as well and thus I think it makes sens to have a separate function for it.

Reply:
A second thought in this, since we want the user to be able to set a env parameter, I suggest we keep this function to have the possibility to check if the query parameter is empty of not and if it is go check for the env parameter.

Reply:
I also suggest keeping it as string and not use ` in order to see if it is set or not since both true and false are valid but if it is "" then we should go for the env parameter.

Reply:
What are your thoughts @Bobgy ?

Reply:
Sorry, I've been busy and didn't have time to go through github mentions.
Regarding this issue, we can also use *bool to represent nil value: https://stackoverflow.com/a/43351386/8745218.

However, in this case, I think your solution of using string is simple enough, so I'm okay with that too.